src/legacy/ui/public/new_platform/new_platform.karma_mock.js

```diff
@@ -469,23 +489,15 @@ export function __setup__(coreSetup) {
   // bootstrap an LP plugin outside of tests)
   npSetup.core.application.register = () => {};
 
-  // Services that need to be set in the legacy platform since the legacy data plugin
-  // which previously provided them has been removed.
-  setInjectedMetadata(npSetup.core.injectedMetadata);
+  // Services that need to be set in the legacy platform since the legacy data
+  // & vis plugins which previously provided them have been removed.
+  setSetupServices(npSetup);
 }
 
 export function __start__(coreStart) {
   npStart.core = coreStart;
 
-  // Services that need to be set in the legacy platform since the legacy data plugin
-  // which previously provided them has been removed.
-  setHttp(npStart.core.http);
-  setNotifications(npStart.core.notifications);
-  setOverlays(npStart.core.overlays);
-  setUiSettings(npStart.core.uiSettings);
-  setFieldFormats(npStart.plugins.data.fieldFormats);
-  setIndexPatterns(npStart.plugins.data.indexPatterns);
-  setQueryService(npStart.plugins.data.query);
-  setSearchService(npStart.plugins.data.search);
-  setAggs(npStart.plugins.data.search.aggs);
+  // Services that need to be set in the legacy platform since the legacy data
+  // & vis plugins which previously provided them have been removed.
+  setStartServices(npStart);
```

Comment:
I did some overall cleanup in this file, because it kept getting messier with the addition of the vis mocks